system_status:
  frame_id:
    type: string
    default_value: built_in_computer
    description: Frame where computer is located.
  hardware_id:
    type: string
    default_value: Built-in Computer
    description: Name of hardware used in diagnostics.
  publish_rate:
    type: double
    default_value: 0.25
    description: System status publish rate in seconds.
    validation: {
      bounds<>: [0.0, 1000.0]
    }
  disk_usage_warn_threshold:
    type: double
    default_value: 95.0
    description: Threshold for disk usage warning in percentage.
    validation: {
      bounds<>: [0.0, 100.0]
    }
  memory_usage_warn_threshold:
    type: double
    default_value: 95.0
    description: Threshold for memory usage warning in percentage.
    validation: {
      bounds<>: [0.0, 100.0]
    }
  cpu_usage_warn_threshold:
    type: double
    default_value: 95.0
    description: Threshold for CPU usage warning in percentage.
    validation: {
      bounds<>: [0.0, 100.0]
    }
  cpu_temperature_warn_threshold:
    type: double
    default_value: 80.0
    description: Threshold for CPU temperature warning in degrees Celsius.
    validation: {
      bounds<>: [0.0, 1000.0]
    }
